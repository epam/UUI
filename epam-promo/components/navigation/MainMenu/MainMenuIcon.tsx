import * as React from 'react';
import { Icon, IAdaptiveItem, cx } from '@epam/uui-core';
import { ButtonProps } from '@epam/uui-components';
import { IconButton } from '../../buttons';
import css from './MainMenuIcon.scss';

export interface MainMenuIconProps extends ButtonProps, IAdaptiveItem {
    icon: Icon;
}

export const MainMenuIcon = React.forwardRef<HTMLButtonElement | HTMLAnchorElement, MainMenuIconProps>((props, ref) => (
    <IconButton ref={ref} icon={props.icon} cx={cx(props.cx, css.container)} {...props} />
));
