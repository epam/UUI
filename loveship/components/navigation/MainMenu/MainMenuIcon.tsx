import React from 'react';
import { Icon, IAdaptiveItem, cx } from '@epam/uui-core';
import { ButtonProps } from '@epam/uui-components';
import { IconButton } from '../../buttons';
import css from './MainMenuIcon.scss';

export interface MainMenuIconProps extends ButtonProps, IAdaptiveItem {
    icon: Icon;
}

export const MainMenuIcon = React.forwardRef<HTMLDivElement, MainMenuIconProps>((props: MainMenuIconProps, ref) => (
    <IconButton ref={ref} icon={props.icon} cx={cx(props.cx, css.container)} {...props} />
));

MainMenuIcon.displayName = 'MainMenuIcon';
