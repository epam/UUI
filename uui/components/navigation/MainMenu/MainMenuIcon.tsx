import * as React from 'react';
import { Icon, IAdaptiveItem, cx } from '@epam/uui-core';
import { ButtonProps } from '@epam/uui-components';
import { IconButton } from '../../buttons';
import css from './MainMenuIcon.module.scss';

/** Represents the properties of the MainMenuIcon component. */
export type MainMenuIconProps = ButtonProps & IAdaptiveItem & {
    /*
    * Defines component icon.
    */
    icon: Icon;
};

export const MainMenuIcon = /* @__PURE__ */React.forwardRef<HTMLButtonElement | HTMLAnchorElement, MainMenuIconProps>((props, ref) => (
    <IconButton ref={ ref } icon={ props.icon } cx={ cx(props.cx, css.container) } { ...props } />
));
