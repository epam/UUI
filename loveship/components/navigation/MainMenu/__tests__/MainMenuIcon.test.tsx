import * as React from 'react';
import renderer from 'react-test-renderer';
import { MainMenuIcon } from '../MainMenuIcon';
import { ReactComponent as AcceptIcon } from '../../../icons/accept-12.svg';

describe('MainMenuIcon', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<MainMenuIcon icon={AcceptIcon} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const tree = renderer
            .create(<MainMenuIcon icon={AcceptIcon} target="_blank" link={{ pathname: '/' }} collapseToMore estimatedWidth={120} priority={6} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
