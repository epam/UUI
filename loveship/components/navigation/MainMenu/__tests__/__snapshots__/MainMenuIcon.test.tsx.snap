// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuIcon should be rendered correctly 1`] = `
<button
  className="container uui-button-box uui-enabled root color-night600 container"
  href={null}
  onClick={[Function]}
  role="button"
  tabIndex={-1}
  type="button"
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
</button>
`;

exports[`MainMenuIcon should be rendered correctly with props 1`] = `
<a
  className="container uui-button-box uui-enabled -clickable root color-night600 container"
  onClick={[Function]}
  role="link"
  tabIndex={0}
  target="_blank"
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
</a>
`;
